lib/reqTemplate.js

```diff
@@ -69,35 +106,46 @@ function splitAndPrepareTAsseblyTemplate(templateSpec, options) {
     var templateNest = 0;
     var startIndex = 0;
     var currentTemplate;
+    var inDoubleBrace = false;
     for (var index = 0; index < templateSpec.length; index++) {
         if (templateSpec[index] === '{') {
             if (templateNest === 0) { // We are either entering a new template
                 if (startIndex !== index) {
                     result.push(templateSpec.substring(startIndex, index));
                 }
+                if (templateSpec[index + 1] === '{') {
+                    index++;
+                    inDoubleBrace = true;
+                }
                 startIndex = index + 1;
+
```

Comment:
Removed this one in a follow-up. Others below are intentional, to group related functionality a bit.